### PR TITLE
Task1 pull request

### PR DESCRIPTION
**Pull request by MinHius**
Task 1 fix report:
- Edge case tests were added according to your comments
- Some redundant and illogical variables and codes were removed, some replaced.
- Expanded todo.py file to contain complex inputs for a thorough test

### CODE REVIEW

File: Extract.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    return result, count, prompt
+        
+
+def full_extraction(extract):
+    count = 0
+    level = 0
+    indent = '   ' * level
+    result = []
+    prompt = []
+    end_imports = True
```
Comment (thanhtung6824) [inline Extract.py:81]: why end_imports start as True?

Comment (thanhtung6824) [reply Extract.py:81]: It should start as False (no imports ended yet)

File: Extract.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            for return_ in ast.walk(node):
+                if isinstance(return_, ast.Return):
+                   result.append(f"return: {type(return_.value).__name__}")
+                   prompt.append(f"{count}|{'   ' * (level + 1)}return: {type(return_.value).__name__}")
+                   count += 1
+  
+    return result, count, prompt
+        
+
+def full_extraction(extract):
```
Comment (thanhtung6824) [inline Extract.py:75]: unused parameter

File: Extract.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    count = 0
+    level = 0
+    indent = '   ' * level
+    result = []
+    prompt = []
+    end_imports = True
+    
+    with open("todo.py", "r") as f:
```
Comment (thanhtung6824) [inline Extract.py:83]: with open(filename, "r") as f

File: Extract.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        args.append(f"**{node.args.kwarg.arg}")
+
+    return f"def {node.name}({', '.join(args)}):"
+
+# Extract classes only
+def class_extraction(extract, count, level, prompt):
+    result = []
+    indent = '   ' * level
+    # Print classes
+    for node in ast.walk(extract):
```
Comment (thanhtung6824) [inline Extract.py:27]: Inefficient. ast.walk() visits every single node in the AST you give, be selective

Comment (thanhtung6824) [reply Extract.py:27]: for example: `full_extraction` visits ALL nodes in TodoClass, `class_extraction` and `func_extraction` walks same node again. O(n^3) complexity instead of O(n)

**Comment by thanhtung6824** [comment]
Test missing edge cases
No tests for:

Empty files
Files with no classes
Files with only imports
Malformed Python code
Functions with complex signatures (*args, **kwargs, defaults)
